prody/proteins/interactions.py

```diff
@@ -0,0 +1,2478 @@
+           'addHydrogens', 'calcHydrogenBondsDCD',
+           'Interactions', 'InteractionsDCD']
+
+
+def cleanNumbers(listContacts):
+    """Provide short list with indices and value of distance"""
+    
+    shortList = [ [int(str(i[0]).split()[-1].strip(')')), 
+                           int(str(i[1]).split()[-1].strip(')')), 
+                           str(i[0]).split()[1], 
+                           str(i[1]).split()[1], 
+                           float(i[2])] for i in listContacts ]    
+    
+    return shortList
+
+
+def calcPlane(atoms):
+    """Function provide parameters of a plane for aromatic rings (based on 3 points).
+    Used in calcPiStacking()"""
+    
+    coordinates = atoms.getCoords()
+    p1, p2, p3 = coordinates[:3] # 3 points will be enough to obtain the plane
+    x1, y1, z1 = p1
+    x2, y2, z2 = p2
+    x3, y3, z3 = p3    
+    vec1 = p3 - p1 # These two vectors are in the plane
+    vec2 = p2 - p1
+    cp = np.cross(vec1, vec2) # the cross product is a vector normal to the plane
+    a, b, c = cp
+    d = np.dot(cp, p3) # This evaluates a * x3 + b * y3 + c * z3 which equals d
+    
+    return a,b,c,d
+
+
+def calcAngleBetweenPlanes(a1, b1, c1, a2, b2, c2):  
+    """Find angle between two planes"""
+    import math 
+          
+    d = ( a1 * a2 + b1 * b2 + c1 * c2 ) 
+    eq1 = math.sqrt( a1 * a1 + b1 * b1 + c1 * c1) 
+    eq2 = math.sqrt( a2 * a2 + b2 * b2 + c2 * c2) 
+    d = d / (eq1 * eq2) 
+    AngleBetweenPlanes = math.degrees(math.acos(d)) 
+    
+    return AngleBetweenPlanes
+    
+    
+def removeDuplicates(list_of_interactions):
+    ls=[]
+    newList = []
+    for no, i in enumerate(list_of_interactions):
+       i = sorted(list(array(i).astype(str)))
+       if i not in ls:
+           ls.append(i)
+           newList.append(list_of_interactions[no])
+    return newList
+
+
+def selectionByKwargs(list_of_interactions, atoms, **kwargs):
+    """Return interactions based on selection"""
+    
+    if 'selection' in kwargs:
+        if 'selection2' in kwargs:
+            ch1 = kwargs['selection'].split()[-1] 
+            ch2 = kwargs['selection2'].split()[-1] 
+            final = [i for i in list_of_interactions if (i[2] == ch1 and i[5] == ch2) or (i[5] == ch1 and i[2] == ch2)]
+        else:
+            p = atoms.select('same residue as protein within 10 of ('+kwargs['selection']+')')
+            x = p.select(kwargs['selection']).getResnames()
+            y = p.select(kwargs['selection']).getResnums()
+            listOfselection = np.unique(list(map(lambda x, y: x + str(y), x, y)))
+            final = [i for i in list_of_interactions if i[0] in listOfselection or i[3] in listOfselection]
+    else:
+        final = list_of_interactions
+    return final
+
+
+def addHydrogens(pdb, method='openbabel', pH=7.0):    
+    """Function will add hydrogens to the protein and ligand structure using Openbabel [NO11]_
+    or PDBFixer with OpenMM.
+    
+    :arg pdb: PDB file name
```

Comment:
we usually use pdb when there's an option to use a pdb id and parse it. Otherwise, it's probably better to call it filename